### PR TITLE
🍺 - Creator Reactfactor

### PR DESCRIPTION
**Pull request by benjdlambert**
### Problem
- The Brands JSX is much easier to follow
- The Creator JSX is all over the place
- The Creator JSX should follow `presentational` and `container` components

### Solution
- Refactor some components to remove the state and push it into container components
- Make the directory tree in the Creator JSX much cleaner

```
|__creator
| |__containers
| |__data
| |__pages
| |__presentational
```

### Notes
@revett @stevenjack

### CODE REVIEW

File: Gulpfile.js [hunk 1]
```diff
@@ -125,7 +125,7 @@ gulp.task(
 
 gulp.task(
   'build:jsx:creator',
-  () => gulp.src('source/jsx/creator/onboarding.jsx')
+  () => gulp.src('source/jsx/creator/pages/onboarding.jsx')
```
Comment (revett) [inline Gulpfile.js]: Why not just:

```js
() => gulp.src('source/jsx/creator/pages/*.jsx')
```

File: source/jsx/creator/presentational/onboarding_collective.jsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+
+export default function Collective(props) {
```
Comment (benjdlambert) [inline source/jsx/creator/presentational/onboarding_collective.jsx]: This is the recommended way to create components that doesn't have any internal state.

File: source/jsx/creator/presentational/onboarding_collective.jsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        <span className="title icon-tick">{props.name}</span>
+        <span className="image">
+          <img role="presentation" src={props.image_src} />
+        </span>
+      </label>
+    </li>
+  );
+}
+
+Collective.propTypes = {
```
Comment (revett) [inline source/jsx/creator/presentational/onboarding_collective.jsx]: 🏄 

Move to using new PropTypes structure

Comment (benjdlambert) [inline source/jsx/creator/presentational/onboarding_collective.jsx]: Can't as it's a `functional` component. It has to be created like this.

File: Gulpfile.js (hunk 1 above)
Comment (benjdlambert) [inline Gulpfile.js]: Was going to leave that till the next PR, as that is the one where its just one app.

Right now, it's just parity as it's a refactor PR.

File: source/jsx/creator/presentational/onboarding_collective.jsx (hunk 3 above)
Comment (benjdlambert) [inline source/jsx/creator/presentational/onboarding_collective.jsx]: Unless we move this to a `class` then we could.
But then we would get eslint of `Useless constructor`.

Which is worse? 🙈

File: Gulpfile.js (hunk 1 above)
Comment (revett) [inline Gulpfile.js]: Yep sounds good

File: source/jsx/creator/presentational/onboarding_collective.jsx (hunk 3 above)
Comment (revett) [inline source/jsx/creator/presentational/onboarding_collective.jsx]: Haha let's keep it as is

**Comment by benjdlambert** [comment]
[[SA] [CR] Refactor Creator React apps to new directory structure](https://trello.com/c/Tc7Du1VI/564-sa-cr-refactor-creator-react-apps-to-new-directory-structure)

**Pull request merged**